### PR TITLE
OLS-39: Remove condition for query validation if it is a follow-up question

### PR DESCRIPTION
**Pull request by onmete**
## Description

Remove condition for query validation if it is a follow-up question

## Type of change

- [x] Refactor
- [x] New feature

### CODE REVIEW

File: ols/app/endpoints/ols.py [hunk 1]
```diff
@@ -120,12 +120,7 @@ def conversation_request(
     validate_requested_provider_model(llm_request)
 
     # Validate the query
-    if not previous_input:
-        valid = validate_question(conversation_id, llm_request)
-    else:
-        logger.debug("follow-up conversation - skipping question validation")
-        valid = True
```
Comment (onmete) [inline ols/app/endpoints/ols.py:127]: TBD - I don't see much sense in having it in code when we agreed we don't want to do that. If anything, this should be part of validation configuration (and not hardcoded) - if only initial question or also follow-ups should be validated.
So I would remove this and once there is a need/usecase for having question validation (besides system instructions), let's do proper configuration.

Comment (onmete) [reply ols/app/endpoints/ols.py:127]: @tisnik @asamal4 WDYT?

Comment (xrajesh) [reply ols/app/endpoints/ols.py:127]: This change does not modify the current behavior as we dont do any validation by default. 
/lgtm

**Comment by tisnik** [comment]
/approve
/override "Red Hat Konflux / ols-enterprise-contract / lightspeed-service"

**Pull request merged**